addons/project/models/project.py

```diff
@@ -1248,6 +1311,43 @@ def write(self, vals):
                 recurrence_domain = [('recurrence_id', 'in', self.recurrence_id.ids)]
             tasks |= self.env['project.task'].search(recurrence_domain)
 
+        # Track fields on depend_on_ids, if a task depends on this task, certain fields have to be tracked in the
+        # `parent` task.
+        if self.user_has_groups('project.group_project_task_dependencies'):
+            dependency_enabled_tasks = self.filtered('allow_task_dependencies')
+            depends_tracked_fields = self._get_depends_tracked_fields()
+            dependent_tasks = None
+            if dependency_enabled_tasks:
+                dependent_tasks = self.env['project.task'].search([
+                    ('allow_task_dependencies', '=', True), ('depend_on_ids', 'in', dependency_enabled_tasks.ids)])
+            # The rest of this code is inspired by `mail.thread` and serves the purpose of creating only one message per change in a commit
+            self.env.cr.precommit.add(self._dependency_finalize_tracking)
```

Comment:
I would rather override ``_mail_track`` to ensure we don't create tracking in case there is something like ``mail_notrack`` in the context, instead of doing it in the write method.

Reply:
And also, by doing this, you can retrieve the tracking values, and avoid calling _mail_track a second time. :)